### PR TITLE
Add options and message size limits

### PR DESCRIPTION
**Pull request by JamesNK**
Addresses https://github.com/grpc/grpc-dotnet/issues/52

### CODE REVIEW

File: src/Grpc.AspNetCore.Server/GrpcServiceExtensions.cs [hunk 1]
```diff
@@ -29,22 +30,59 @@ namespace Microsoft.Extensions.DependencyInjection
...
+
+        /// <summary>
+        /// Adds gRPC services to the specified <see cref="IServiceCollection" />.
         /// </summary>
         /// <param name="services">The <see cref="IServiceCollection"/> for adding services.</param>
-        /// <returns></returns>
-        // TODO(JunTaoLuo): Options?
-        public static IServiceCollection AddGrpc(this IServiceCollection services)
+        /// <returns>An <see cref="IGrpcServerBuilder"/> that can be used to further configure the gRPC services.</returns>
+        public static IGrpcServerBuilder AddGrpc(this IServiceCollection services)
```
Comment (JunTaoLuo) [inline src/Grpc.AspNetCore.Server/GrpcServiceExtensions.cs]: Why do we need a separate Builder type?

Comment (JamesNK) [reply src/Grpc.AspNetCore.Server/GrpcServiceExtensions.cs]: https://github.com/grpc/grpc-dotnet/blob/43cd16735b633e6b0d8cdc5837dc48726b87291d/test/Grpc.AspNetCore.Server.Tests/GrpcServicesExtensionsTests.cs#L64-L67

Comment (JunTaoLuo) [reply src/Grpc.AspNetCore.Server/GrpcServiceExtensions.cs]: Can this not be done through just a ServiceCollection? It looks like IGrpcServerBuilder is just a wrapper around it anyway. Is there more things you want to add to the interface later?

Comment (JamesNK) [reply src/Grpc.AspNetCore.Server/GrpcServiceExtensions.cs]: If it was on a ServiceCollection then you could call AddServiceOptions without AddGrpc, which doesn't make sense. It is there for intelli-sense. MVC and SignalR do the same thing.

Comment (JunTaoLuo) [reply src/Grpc.AspNetCore.Server/GrpcServiceExtensions.cs]: Sounds good

File: src/Grpc.AspNetCore.Server/Internal/ServerCallHandlerBase.cs [hunk 2]
```diff
@@ -26,10 +26,12 @@ namespace Grpc.AspNetCore.Server.Internal
     internal abstract class ServerCallHandlerBase<TRequest, TResponse, TService> : IServerCallHandler
     {
         protected Method<TRequest, TResponse> Method { get; }
+        protected GrpcServiceOptions ServiceOptions { get; }
 
-        protected ServerCallHandlerBase(Method<TRequest, TResponse> method)
+        protected ServerCallHandlerBase(Method<TRequest, TResponse> method, GrpcServiceOptions serviceOptions)
         {
-            Method = method ?? throw new ArgumentNullException(nameof(method));
+            Method = method;
```
Comment (jtattermusch) [inline src/Grpc.AspNetCore.Server/Internal/ServerCallHandlerBase.cs]: nit: can the method be null now?

**Pull request merged**